### DISCUSSION TITLE
Method for resetting admin password

### DISCUSSION THREAD

**Discussion by CorvusCallosum** [felddy/foundryvtt-docker#738] [Q&A] [answered]
I must be missing something with how the configuration works. I returned to the server after not much use and ran the latest update to Foundry VTT 11. Now I don't know the admin password and clearing the config file (https://foundryvtt.com/article/reset-admin-password/) doesn't seem to do it, nor does removing the container and rebuilding. Is there a recommended method for this? Thanks.

**Comment by felddy** [comment, marked as answer]
The reason clearing the file doesn't work is because the file is [generated (or regenerated) at container startup](https://github.com/felddy/foundryvtt-docker#configuration-management).   This means you probably have set the admin password using either environment variables or [using secrets](https://github.com/felddy/foundryvtt-docker#using-secrets).  

Check to see if you have set the `FOUNDRY_ADMIN_KEY` environment variable, or passed in a secrets file that set the `foundry_admin_key` key.

**Reply by CorvusCallosum** [reply]
Thank you for the quick reply. It was created using secrets. 

I got around the issue by commenting out the secrets in my docker compose file, deleting the config file, and restarting the container. Then I had to paste my key back in from the FoundryVTT website. After reverting the compose file to contain the secrets again, things are working. Let me know if this is not a recommended method, but it was pretty quick.

**Reply by felddy** [reply]
That'll work.  You could have also looked at your `secrets.json` file which would have revealed the admin password to you.  

In any case, I'm glad you're back up and running.  Huzzah! 🎉

**Reply by CorvusCallosum** [reply]
Really? Those are different for me. My secrets.json contains my https://foundryvtt.com/ login info and license key. The login info there doesn't work to sign in as an admin on my foundry server.

**Reply by felddy** [reply]
Yup.  Currently there are five (5) different keys you can pass in through secrets.  The `foundry_admin_key` is one of them.  

See:
- https://github.com/felddy/foundryvtt-docker#secrets

**Reply by CorvusCallosum** [reply]
Ah ok. So I could have included the admin key but didn't. My incomplete setup made it more complicated than it needed to be to reset. Thanks again.
